pkg/odo/genericclioptions/context.go

```diff
@@ -360,8 +369,11 @@ func newContext(command *cobra.Command, createAppIfNeeded bool, ignoreMissingCon
 		util.LogErrorAndExit(err, "")
 	}
 
+	namespace := "default"
 	// Resolve project
-	namespace := resolveProject(command, client, localConfiguration)
+	if client != nil {
+		namespace = resolveProject(command, client, localConfiguration)
```

Comment:
wouldn't it be possible to get the namespace from the localConfig if the client in nil?